### ISSUE TITLE
NPM package.json formatter

### ISSUE THREAD

**Issue by muuvmuuv** [Unibeautify/unibeautify#206]
**Beautifier Details**
Name: sort-package-json
Link to project/homepage: https://github.com/keithamus/sort-package-json

**If this is a new beautifier, please fill out the below**
Languages Supported: package.json (only)
How is it run (node, terminal, etc): node

**Additional context**

I think this formatter could be a good thing to keep package.json files formatted correctly in all projects or everytime someone edit them manually. I use it daily to keep them clean when "Pretty Diff" is formatting them a little bit different.

I think @Glavin001 commented some day that I should keep the line break in the arrays, this would fix this problem, because Pretty Diff always keep them on one line.

```json
{
  "dependencies": {},
  "version": "1.0.0",
  "keywords": ["thing"],
  "name": "foo",
}
```
after
```json
{
  "name": "foo",
  "version": "1.0.0",
  "keywords": [
    "thing"
  ],
  "dependencies": {}
}
```

**Comment by muuvmuuv** [comment]
A possible beautifier name could be _beautifier-npm_, I think _beautifier-sort-package-json_ is a little bit too long, but I can understand if we want to keep it consistent.

**Comment by stevenzeck** [comment]
We generally want to keep the names the same so it doesn't confuse anyone.  Does it format as well as sort?

**Comment by muuvmuuv** [comment]
@stevenzeck I need to check that. I guess it just sorts it. If not we could add a note for users to use it with prettier behind to beautify it first

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Glavin001** [comment]
Prettier actually supports `JSON.stringify`-like formatting: https://github.com/prettier/prettier/pull/4450
Uses `json-stringify` parser: https://github.com/prettier/prettier/blob/808ce8817b20eb016f404a03a7c822f4353660e4/src/language-js/index.js#L131-L144

**Comment by muuvmuuv** [comment]
@Glavin001 yes and Sort package would sort it not beautify it actually. I guess people can use it without Unibeautify but I thought it would be nice to have this on save

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Glavin001** [comment]
@muuvmuuv they could be two separate beautifiers. Improve the existing Prettier beautifier to better support `package.json` and also create a new beautifier for sorting `package.json` keys. Then in `.unibeautifyrc` file, have both `beautifers: ["Prettier", "Sort-Package-JSON"]` for example. 🎉

**Comment by lassik** [comment]
FYI: https://github.com/prettier/prettier/issues/1096 Option to sort object keys

"Currently out of scope" for prettier.

EDIT: That issue seems to be about keys in JavaScript code, where the values of those keys can be executable code. JSON would be easier as order of keys cannot affect semantics.

**Comment by muuvmuuv** [comment]
@lassik I do not think this is the same. `sort-package-json` sort by the well-known keys from npm see https://www.npmjs.com/package/sort-package-json#how-does-it-sort

@Glavin001 I would take on this if we are going to create a beautifier for `sort-package-json`

**Comment by lassik** [comment]
OK, so it doesn't sort them in alphabetical order, but a custom one. Cool.
